File: AlgorithmsPackCPP/Find Right Interval.cpp
class Solution {
public:
    vector<int> findRightInterval(vector<vector<int>>& intervals) {
        map<int,int>startsToIndices;
        int n = intervals.size();
        for(int i = 0; i < n; i++){
            startsToIndices[intervals[i][0]]=i;
        }
        vector<int>result(n,-1);
        for(int i = 0; i < n; i++){
            auto iter = startsToIndices.lower_bound(intervals[i][1]);
            if(iter!=startsToIndices.end())
                result[i] = iter->second;
        }
        return result;
    }
};

File: AlgorithmsPackPythonSolutions/Find Right Interval.py
class Solution:
    def findRightInterval(self, intervals):

        startsToIndices = {}
        for i in range(len(intervals)):
            startsToIndices[intervals[i][0]] = i
            
        result = [-1] * len(intervals)
        for i in range(len(intervals)):
            start = intervals[i][1]
            if start in startsToIndices:
                result[i] = startsToIndices[start]
            else:
                keys = sorted(startsToIndices.keys())
                index = bisect.bisect_right(keys, start)
                if index < len(keys):
                    result[i] = startsToIndices[keys[index]]
        
        return result
